src/doc/rustc/src/codegen-options/index.md

```diff
@@ -553,9 +553,12 @@ Supported values for this option are:
   of MSVC).
 - `debuginfo` - debuginfo sections and debuginfo symbols from the symbol table
   section are stripped at link time and are not copied to the produced binary
-  or separate files.
+  or separate files. This should leave backtraces mostly-intact but may make
+  using a debugger like gdb or lldb ineffectual.
 - `symbols` - same as `debuginfo`, but the rest of the symbol table section is
-  stripped as well if the linker supports it.
+  stripped as well if the linker supports it. On platforms which depend on the
+  binary's symbol table for backtraces, this can affect them so negatively as to
+  make the trace completely incomprehensible.
```

Comment:
can you also mention that it may also make the binary uninstrumentable in general? maybe even discourage it - for example I don't think a cli binary expected to run on a desktop computer should ever set this

Reply:
@Nilstrieb I've never heard of it simply making the binary uninstrumentable! What examples were you thinking of?

Reply:
Ah this was about `perf`! I see.

Reply:
just anything really. by stripping symbols, you make the binary entirely opaque. it is impossible to know what's going on in this binary afterwards (assuming no debug info, which usually isn't present in stripped Rust programs).

Reply:
right.

Reply:
How does it look now?

Reply:
Great!